Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -12,137 +13,171 @@ struct NestingRule: Rule {
+            guard !configuration.ignoreTypealiasesAndAssociatedtypes else { return }
+            validate(node, triggeringToken: node.associatedtypeKeyword, inFunction: false)
+            validationData.pop()
+        }
+
+        // MARK: - configuration for checkNestingInClosuresAndStatements
+        override func visit(_ node: ClosureExprSyntax) -> SyntaxVisitorContinueKind {
+            if !configuration.checkNestingInClosuresAndStatements {
+                return .skipChildren
+            } else {
+                return super.visit(node)
+            }
         }
 
-        guard let severity = violatingSeverity else {
-            return nil
+        override func visit(_ node: CodeBlockItemSyntax) -> SyntaxVisitorContinueKind {
+            if !configuration.checkNestingInClosuresAndStatements && node.parent?.inStatement ?? false {
+                return .skipChildren
+            } else {
+                return super.visit(node)
+            }
         }
 
-        let targetName = forFunction ? "Functions" : "Types"
-        let threshold = configuration.threshold(with: targetLevel, for: severity)
-        let pluralSuffix = threshold > 1 ? "s" : ""
-        return StyleViolation(
-            ruleDescription: Self.description,
-            severity: severity,
-            location: Location(file: file, byteOffset: offset),
-            reason: "\(targetName) should be nested at most \(threshold) level\(pluralSuffix) deep"
-        )
+        // MARK: -
+        private func validate(_ node: some DeclSyntaxProtocol, triggeringToken: TokenSyntax? = nil, inFunction: Bool) {
+            validationData.push(node)
+            let isFunction = validationData.isFunction
+            let (level, targetLevel) = if isFunction {
+                (validationData.functionLevel, configuration.functionLevel)
+            } else {
+                (validationData.typeLevel, configuration.typeLevel)
+            }
+
+            var violatingSeverity: ViolationSeverity?
+
+            if configuration.alwaysAllowOneTypeInFunctions, !inFunction {
+                violatingSeverity = configuration.severity(with: targetLevel, for: level)
+            } else if isFunction || !configuration.alwaysAllowOneTypeInFunctions {
+                violatingSeverity = configuration.severity(with: targetLevel, for: level)
+            } else {
+                violatingSeverity = nil
+            }
+
+            guard let severity = violatingSeverity else {
+                return
+            }
+
+            let targetName = isFunction ? "Functions" : "Types"
+            let threshold = configuration.threshold(with: targetLevel, for: severity)
+            let pluralSuffix = threshold > 1 ? "s" : ""
+            let position = (triggeringToken?.positionAfterSkippingLeadingTrivia
+                            ?? node.positionAfterSkippingLeadingTrivia)
+            violations.append(.init(
```

Comment:
Please specify the type:
```suggestion
            violations.append(ReasonedRuleViolation(
```